CMSIS/Core/Include/cachel1_armv7.h

```diff
@@ -181,30 +181,53 @@ __STATIC_FORCEINLINE void SCB_EnableDCache (void)
 __STATIC_FORCEINLINE void SCB_DisableDCache (void)
 {
   #if defined (__DCACHE_PRESENT) && (__DCACHE_PRESENT == 1U)
-    uint32_t ccsidr;
-    uint32_t sets;
-    uint32_t ways;
+    struct {
+      uint32_t ccsidr;
+      uint32_t sets;
+      uint32_t ways;
+    } locals
+    #if (defined(__GNUC__) && !defined(__OPTIMIZE__))
+       __attribute__((aligned(__SCB_DCACHE_LINE_SIZE)))
```

Comment:
This attribute will have to be replaced with `__ALIGNED(__SCB_DCACHE_LINE_SIZE)` from the CMSIS compiler polyfill definitions (`cmsis_compiler.h` and its includes). Even though it's currently only for gcc, it should match the rest of CMSIS. Sorry, I should have used it in my example, but was just writing it quickly in the browser…

Speaking of, does this issue really only affect gcc? What about clang, armclang, IAR, etc?

Reply:
IAR and Keil are not affected. When disabling optimization on those compilers they still do some minor optimization which leads to placing Variables in registers only instead of using the stack.
Disabling optimization on the GCC means that there is really no optimization, and the variables are placed on the stack. I’ve done some tests with clang and it behaves like GCC.
I’ve updated the PR to fix also the clang issue. The __ALIGNED macro is also used.

Reply:
Thanks for the updates!